lib/internal/modules/esm/loader.js

```diff
@@ -130,13 +129,10 @@ class Loader {
   }
 
   hook({ resolve, dynamicInstantiate }) {
-    // Use .bind() to avoid giving access to the Loader instance when called.
```

Comment:
Is the original intent still respected?

Reply:
Yea, hence the other change above which ensures that the callbacks aren't called within the loader scope.